src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx-stm32f.inl.h

```diff
@@ -0,0 +1,445 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IOTJS_MODULE_GPIO_ARM_NUTTX_STM32F_INL_H
+#define IOTJS_MODULE_GPIO_ARM_NUTTX_STM32F_INL_H
+
+#include "stm32_gpio.h"
+#include "module/iotjs_module_gpio.h"
+
+#define GPIO_FREQUENCY_DEFAULT GPIO_SPEED_25MHz
+#define GPIO_PINCNT_IN_NUTTXPORT 16
+#define GPIO_PORTCNT_IN_NUTTXPORT 2
+
+#define GPIO_CONFIG_OK 0
+
+// Change iotjs pin or port number to nuttx gpio pin or port index
+static inline int32_t GetGpioPortIndexFromPinNumber(int32_t pin) {
+  return (pin / GPIO_PINCNT_IN_NUTTXPORT);
+}
+
+
+static inline int32_t GetGpioPinIndexFromPinNumber(int32_t pin) {
+  return (pin % GPIO_PINCNT_IN_NUTTXPORT);
+}
+
+
+static inline int32_t GetGpioPortIndexFromPortNumber(int32_t port) {
+  return (port / GPIO_PORTCNT_IN_NUTTXPORT);
+}
+
+
+static inline int32_t GetGpioPinIndexFromPortNumber(int32_t port) {
+  return ((port % GPIO_PORTCNT_IN_NUTTXPORT) * GPIO_PINCNT_IN_IOTJS);
+}
+
+
+uint32_t gpioDirection[3] = {
+  0,  // none
+  GPIO_INPUT,
+  GPIO_OUTPUT | GPIO_OUTPUT_SET | GPIO_FREQUENCY_DEFAULT
+};
+
+
+uint32_t gpioPort[7] = {
+  GPIO_PORTA,
+  GPIO_PORTB,
+  GPIO_PORTC,
+  GPIO_PORTD,
+  GPIO_PORTE,
+  GPIO_PORTF,
+  GPIO_PORTG
+};
+
+
+uint32_t gpioPin[GPIO_MAX_PINNO] = {
+  GPIO_PIN0,
+  GPIO_PIN1,
+  GPIO_PIN2,
+  GPIO_PIN3,
+  GPIO_PIN4,
+  GPIO_PIN5,
+  GPIO_PIN6,
+  GPIO_PIN7,
+  GPIO_PIN8,
+  GPIO_PIN9,
+  GPIO_PIN10,
+  GPIO_PIN11,
+  GPIO_PIN12,
+  GPIO_PIN13,
+  GPIO_PIN14,
+  GPIO_PIN15
+};
+
+
+uint32_t gpioMode[6] = {
+  0,  // none
+  GPIO_PULLUP,
+  GPIO_PULLDOWN,
+  GPIO_FLOAT,
+  GPIO_PUSHPULL,
+  GPIO_OPENDRAIN
+};
+
+
+namespace iotjs {
+
+
+// GPIO implementeation for arm-nuttx target.
+class GpioArmNuttxStm32f : public Gpio {
+ public:
+  explicit GpioArmNuttxStm32f(JObject& jgpio);
+
+  static GpioArmNuttxStm32f* GetInstance();
+
+  virtual int Initialize(GpioReqWrap* gpio_req);
+  virtual int Release(GpioReqWrap* gpio_req);
+  virtual int SetPin(GpioReqWrap* gpio_req);
+  virtual int WritePin(GpioReqWrap* gpio_req);
+  virtual int ReadPin(GpioReqWrap* gpio_req);
+  virtual int SetPort(GpioReqWrap* gpio_req);
+  virtual int WritePort(GpioReqWrap* gpio_req);
+  virtual int ReadPort(GpioReqWrap* gpio_req);
+
+  bool _initialized;
+};
+
+
+GpioArmNuttxStm32f::GpioArmNuttxStm32f(JObject& jgpio)
+    : Gpio(jgpio)
+    , _initialized(false) {
+}
+
+
+GpioArmNuttxStm32f* GpioArmNuttxStm32f::GetInstance()
+{
+  return static_cast<GpioArmNuttxStm32f*>(Gpio::GetInstance());
+}
+
+
+#define GPIO_WORKER_INIT_TEMPLATE \
+  GpioArmNuttxStm32f* gpio = GpioArmNuttxStm32f::GetInstance(); \
+  IOTJS_ASSERT(gpio->_initialized == false); \
+  GpioReqWrap* gpio_req = reinterpret_cast<GpioReqWrap*>(work_req->data); \
+  GpioReqData* req_data = gpio_req->req(); \
+
+
+void AfterWork(uv_work_t* work_req, int status) {
```

Comment:
Seeing I2C module, in order to make correspondency how abour renaming it to ``AfterGPIOWork``?